## SMR locus plot

### Visualization of SMR results

Here we provide an R script to plot SMR results as presented in Zhu et
al. ([2016 Nature Genetics](http://www.nature.com/ng/journal/vaop/ncurrent/full/ng.3538.html)).
The data file for plot can be generated by the command below. The R
script is avaliable at [Download](#Download).

#### \# SMR command line to generate a data file for plot

```
smr --bfile mydata --gwas-summary mygwas.ma --beqtl-summary myeqtl --out myplot --plot --probe ILMN_123 --probe-wind 500 --gene-list glist-hg19 
```

**\--plot** saves the data file for plot in text format .

**\--gene-list** specifies a gene range list.

***glist-hg19 (without strand information)***

```
19	58858171	58864865	A1BG
19	58863335	58866549	A1BG-AS1
10	52559168	52645435	A1CF
......
```
This is a text file without headers. The columns are chromosome code,
start of gene, end of gene and gene ID. 

The gene range lists ( mirrored from
[[PLINK2]](https://www.cog-genomics.org/plink2/resources#genelist)
website) are

-   **hg18:**
[glist-hg18](https://www.cog-genomics.org/static/bin/plink/glist-hg18)
([older, ASCII-sorted PNGU
version](http://pngu.mgh.harvard.edu/~purcell/plink/dist/glist-hg18))
-   **hg19:**
[glist-hg19](https://www.cog-genomics.org/static/bin/plink/glist-hg19)
-   **hg38:**
[glist-hg38](https://www.cog-genomics.org/static/bin/plink/glist-hg38)

Not all the genes in the region but only those in the .epi file will be
drawn in the locus plot since the strand information is absent from the
gene list provied in the PLINK2 website. To fix this problem, we updated
SMR to read a gene list with/without strand information.

***glist-hg19 (with strand information)***
```
19	58858171	58864865	A1BG	-
19	58863335	58866549	A1BG-AS1	+
10	52559168	52645435	A1CF	-
......	
```
This is a text file without headers. The columns are chromosome code,
start of gene, end of gene, gene ID and gene strand.

#### \# R commands to draw the plots
```

source("plot_SMR.r") 
# Read the data file in R:
SMRData = ReadSMRData("myplot.ILMN_123.txt")
# Plot the SMR results in a genomic region centred around a probe:
SMRLocusPlot(data=SMRData, smr_thresh=8.4e-6, heidi_thresh=0.05, plotWindow=1000, max_anno_probe=16)
# smr_thresh: genome-wide significance level for the SMR test.
# heidi_thresh: threshold for the HEIDI test. The default value is 0.05.
# cis_wind: size of a window centred around the probe to select cis-eQTLs for plot. The default value is 2000Kb.
# max_anno_probe: maximum number of probe names to be displayed on the figure. The default value is 16.
```
![alt text](./download/smr.regional.png "regional")

```
# Plot effect sizes from GWAS against those from eQTL study:
SMREffectPlot(data=SMRData, trait_name="BMI") 
# trait_name: name of the trait or disease. 

```
![alt text](./download/smr.effect.size.png "effectsize")


