## Options Reference 

<table width="1000" border="0" style="font-family:arial; font-size:14px; align-self:center" >
<tr>
<td align="left"><b>Option</b></td>        
<td align="left"><b>Description</b></td>        
</tr>
<tr>                           
<td align="left"><font color="#993333">--beqtl-flist</font></td>
<td align="left">reads a file to get the full paths of the BESD files.</td>                
</tr> 
<tr>    
<td align="left"><font color="#993333">--beqtl-summary</font></td>
<td align="left">reads summary-level data from a eQTL study in binary format.</td>                 
</tr>
<tr>                           
<td align="left"><font color="#993333">--bfile</font></td>
<td align="left">reads individual-level SNP genotype data in PLINK binary format.</td>                
</tr> 
<tr>                           
<td align="left"><font color="#993333">--bolt-assoc-format</font></td>
<td align="left">reads eQTL summary data in BOLT_LMM output format.</td>                
</tr> 
<tr>                           
<td align="left"><font color="#993333">--cis-wind</font></td>
<td align="left">defines a window centred around the probe to select cis-eQTLs.</td>                
</tr>     
<tr>                           
<td align="left"><font color="#993333">--eqtl-flist</font></td>
<td align="left">reads a file to get probe information and locations of the eQTL summary data files.</td>                
</tr> 
<tr>                           
<td align="left"><font color="#993333">--exclude-exposure-probe</font></td>
<td align="left">excludes a subset of exposure probes from analysis.</td>                
</tr> 
<tr>                           
<td align="left"><font color="#993333">--exclude-outcome-probe</font></td>
<td align="left">excludes a subset of outcome probes from analysis.</td>                
</tr> 
<tr>                           
<td align="left"><font color="#993333">--exclude-probe</font></td>
<td align="left">excludes a subset of probes from analysis.</td>                
</tr>  
<tr>                           
<td align="left"><font color="#993333">--exclude-single-exposure-probe</font></td>
<td align="left">excludes a single exposure probe from analysis.</td>                
</tr> 
<tr>                           
<td align="left"><font color="#993333">--exclude-single-outcome-probe</font></td>
<td align="left">excludes a single outcome probe from analysis.</td>                
</tr> 
<tr>                           
<td align="left"><font color="#993333">--exclude-snp </font></td>
<td align="left">excludes a subset of SNPs from analysis.</td>                
</tr> 
<tr>                           
<td align="left"><font color="#993333">--extract-cis</font></td>
<td align="left">extracts the cis-eQTLs.</td>                
</tr> 
<tr>                           
<td align="left"><font color="#993333">--extract-exposure-probe</font></td>
<td align="left">extracts a subset of exposure probes for analysis.</td>                
</tr> 
<tr>                           
<td align="left"><font color="#993333">--extract-outcome-probe</font></td>
<td align="left">extracts a subset of outcome probes for analysis.</td>                
</tr> 
<tr>                           
<td align="left"><font color="#993333">--extract-probe</font></td>
<td align="left">extracts a subset of probes for analysis.</td>                
</tr> 
<tr>                           
<td align="left"><font color="#993333">--extract-single-exposure-probe</font></td>
<td align="left">extracts a single exposure probe for analysis.</td>                
</tr> 
<tr>                           
<td align="left"><font color="#993333">--extract-single-outcome-probe</font></td>
<td align="left">extracts a single outcome probe for analysis.</td>                
</tr> 
<tr>                           
<td align="left"><font color="#993333">--extract-snp</font></td>
<td align="left">extracts a subset of SNPs for analysis.</td>                
</tr> 
<tr>                           
<td align="left"><font color="#993333">--from-probe </font></td>
<td align="left">specifies the start probe.</td>                
</tr> 
<tr>                           
<td align="left"><font color="#993333">--from-probe-kb</font></td>
<td align="left">specifies the start physical position of the probes.</td>                
</tr> 
<tr>                           
<td align="left"><font color="#993333">--from-snp</font></td>
<td align="left">specifies the start SNP.</td>                
</tr>  
<tr>                           
<td align="left"><font color="#993333">--from-snp-kb</font></td>
<td align="left"> specifies the start physical position of the region.</td>                
</tr>
<tr>                           
<td align="left"><font color="#993333">--gemma-fomat</font></td>
<td align="left">reads eQTL summary data in GEMMA association output format.</td>                
</tr> 
<tr>                           
<td align="left"><font color="#993333">--gene</font></td>
<td align="left">specifies a single gene to select probes.</td>                
</tr> 
<tr>                           
<td align="left"><font color="#993333">--genes</font></td>
<td align="left">extracts a subset of probes which tag the genes in the list.</td>                
</tr>
<tr>                           
<td align="left"><font color="#993333">--gene-list</font></td>
<td align="left">specifies a gene annotation file.</td>                
</tr>
<tr>                           
<td align="left"><font color="#993333">--geno-uni</font></td>
<td align="left">indicates all the input .esd files are identical.</td>                
</tr>
<tr>                           
<td align="left"><font color="#993333">--gwas-summary</font></td>
<td align="left">reads summary-level data from GWAS in GCTA-COJO format.</td>                
</tr> 
<tr>                           
<td align="left"><font color="#993333">--heidi-m</font></td>
<td align="left">minimum requirement of the number of eQTLs used in the HEIDI test.</td>                
</tr> 
<tr>                           
<td align="left"><font color="#993333">--heidi-mtd</font></td>
<td align="left">specify a method for HEIDI test.</td>                
</tr> 
<tr>                           
<td align="left"><font color="#993333">--heidi-off</font></td>
<td align="left">turns off the HEIDI test.</td>                
</tr>
<tr>                           
<td align="left"><font color="#993333">--keep</font></td>
<td align="left">includes a subset of individuals in the reference sample for analysis.</td>                
</tr>   
<tr>                           
<td align="left"><font color="#993333">--ld-pruning</font></td>
<td align="left">LD r-squared threshold for pruning SNPs (eQTLs) in HEIDI test, removing SNPs in high LD with the top associated eQTL.</td>                
</tr>    
<tr>                           
<td align="left"><font color="#993333">--maf</font></td>
<td align="left">removes SNPs based on a minor allele frequency (MAF) threshold in the reference sample.</td>                
</tr>
<tr>                           
<td align="left"><font color="#993333">--make-besd</font></td>
<td align="left">saves summary data in BESD format. By default, the data will be stored in sparse BESD format.</td>                
</tr>  
<tr>                           
<td align="left"><font color="#993333">--make-besd-dense</font></td>
<td align="left">saves summary data of all SNPs for all probes.</td>                
</tr> 
<tr>                           
<td align="left"><font color="#993333">--out</font></td>
<td align="left">specifies filename prefix for output files.</td>                
</tr> 
<tr>                           
<td align="left"><font color="#993333">--p-technical</font></td>
<td align="left">reads a p-value threshold to select technical eQTLs.</td>                
</tr>
<tr>                           
<td align="left"><font color="#993333">--peqtl-heidi</font></td>
<td align="left">threshold of eQTL p-value to select eQTLs for the HEIDI test.</td>                
</tr> 
<tr>                           
<td align="left"><font color="#993333">--peqtl-other</font></td>
<td align="left">threshold of eQTL p-value to select eQTLs apart from the cis and trans regions.</td>                
</tr> 
<tr>                           
<td align="left"><font color="#993333">--peqtl-smr</font></td>
<td align="left">threshold of eQTL p-value to select the top associated eQTL for the SMR test.</td>                
</tr> 
<tr>                           
<td align="left"><font color="#993333">--peqtl-trans </font></td>
<td align="left">threshold of eQTL p-value for trans-SNPs.</td>                
</tr> 
<tr>                           
<td align="left"><font color="#993333">--plink-qassoc-fomat</font></td>
<td align="left">reads eQTL summary data in PLINK-qassoc format.</td>                
</tr>
<tr>                           
<td align="left"><font color="#993333">--plot</font></td>
<td align="left">saves in text format the data for plot.</td>                
</tr>
<tr>                           
<td align="left"><font color="#993333">--probe</font></td>
<td align="left">specifies a single probe.</td>                
</tr> 
<tr>                           
<td align="left"><font color="#993333">--probe-chr</font></td>
<td align="left">specifies a chromosome to select probes.</td>                
</tr> 
<tr>                           
<td align="left"><font color="#993333">--probe-wind</font></td>
<td align="left">defines a window centred on a specified probe.</td>                
</tr>
<tr>                           
<td align="left"><font color="#993333">--qfile</font></td>
<td align="left">reads eQTL summary data in SMR query output format.</td>                
</tr>
<tr>                           
<td align="left"><font color="#993333">--query</font></td>
<td align="left">saves in text format a subset of the eQTL summary dataset based on the specified eQTL p-value threshold.</td>                
</tr>  
<tr>                           
<td align="left"><font color="#993333">--remove</font></td>
<td align="left">excludes a subset of individuals in the reference sample from the analysis.</td>                
</tr> 
<tr>                           
<td align="left"><font color="#993333">--rm-technical</font></td>
<td align="left">specifies the probe hybridization region and excludes the technical eQTLs.</td>                
</tr> 
<tr>                           
<td align="left"><font color="#993333">--set-wind</font></td>
<td align="left">defines a window width (Kb) centred around the top associated cis-eQTL to select SNPs in the cis-region.</td>                
</tr>
<tr>                           
<td align="left"><font color="#993333">--smr-multi</font></td>
<td align="left">turns on set-based SMR test in the cis-region.</td>                
</tr>      
<tr>                           
<td align="left"><font color="#993333">--snp</font></td>
<td align="left">specifies a single SNP.</td>                
</tr> 
<tr>                           
<td align="left"><font color="#993333">--snp-chr</font></td>
<td align="left">specifies a chromosome to select SNPs.</td>                
</tr> 
<tr>                           
<td align="left"><font color="#993333">--to-probe</font></td>
<td align="left">specifies the end probe.</td>                
</tr> 
<tr>                           
<td align="left"><font color="#993333">--to-snp</font></td>
<td align="left">specifies the end SNP.</td>                
</tr>
<tr>                           
<td align="left"><font color="#993333">--snp-wind</font></td>
<td align="left">defines a window centred on a specified SNP.</td>                
</tr>
<tr>                           
<td align="left"><font color="#993333">--to-probe-kb</font></td>
<td align="left">specifies the end physical position of the probes.</td>                
</tr> 
<tr>                           
<td align="left"><font color="#993333">--to-snp-kb</font></td>
<td align="left">specifies the end physical position of the region.</td>                
</tr> 
<tr>                           
<td align="left"><font color="#993333">--trans</font></td>
<td align="left">turns on SMR and HEIDI tests in trans regions.</td>                
</tr> 
<tr>                           
<td align="left"><font color="#993333">--trans-wind</font></td>
<td align="left">defines a window centred around the top associated trans-eQTL to select SNPs .</td>                
</tr> 
<tr>                           
<td align="left"><font color="#993333">--target-snp</font></td>
<td align="left">specifies a SNP as the target for the SMR and HEIDI tests as described above.</td>                
</tr> 
<tr>                           
<td align="left"><font color="#993333">--thread-num</font></td>
<td align="left">specifies the number of OpenMP threads for parallel computing.</td>                
</tr>             
<tr>                           
<td align="left"><font color="#993333">--update-freq </font></td>
<td align="left">reads allele frequency file.</td>                
</tr> 
</table>

